platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/FilterInjector.java

```diff
@@ -78,13 +94,60 @@ public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo
     }
   }
 
+  public void init() {
+    executorService.schedule(this::checkForMissedServletContexts, 1, TimeUnit.SECONDS);
+  }
+
+  public void destroy() {
+    executorService.shutdown();
+  }
+
+  private void checkForMissedServletContexts() {
+    try {
+      BundleContext context = getContext();
+      if (context == null) {
+        return; // bundle is probably refreshing
+      }
+      Collection<ServiceReference<ServletContext>> references =
+          getContext().getServiceReferences(ServletContext.class, null);
+      for (ServiceReference<ServletContext> reference : references) {
+        Bundle refBundle = reference.getBundle();
+        BundleContext bundlectx = refBundle.getBundleContext();
+        ServletContext service = bundlectx.getService(reference);
+
+        if (service.getFilterRegistration(DELEGATING_FILTER) == null) {
+          LOGGER.error(
+              "Platform filter delegate failed to start in time to inject itself into {} {}. This means the {} servlet will not properly attach the user subject to requests. Attempting to resolve the issue by restarting the bundle.",
+              refBundle.getSymbolicName(),
+              refBundle.getBundleId(),
+              refBundle.getSymbolicName());
+
+          // Restarting the missed servlet context bundle so the injector will be able to perform
+          // its job.
+          refBundle.stop();
```

Comment:
✏️ I'd have to spend a bit more time digging about to remember how the OSGi services/bundles behave, but these two lines concern me. My concern is that on one pass through we get to here and successfully call `stop` on the bundle but throw an exception when attempting to restart it. If my memory is correct, we won't find the service again on the next pass through the `executorService`'s loop so we will never attempt to restart it.

Assuming my recollection is correct, we should keep a running cache of any bundles that we stopped that didn't restart - wrapping the `start` call in its own try/catch with an `add` to a set would be sufficient - and check those on these executed loops.

Reply:
I suppose that's possible, but should only happen in the case where a bundle came up that was supposed to have the injected filter where the user then tried to stop the bundle inside the next 60 seconds for some reason.

Not impossible, but I believe a vanishingly small likelihood.

Reply:
I just dislike having one of those.. "oh just make sure you don't stop it too fast or you might put the system in a wierd state and have to restart" kind of things added onto to our list of *meh*. But hey I mean at least this is one we **know** about :D

Reply:
My thought on this was if we are here the system is already in a bad state. If we stop a bundle and it then fails to come up we are in no worse a state than when we started. In the case of the bundle not coming back up it would actually be easier to see something is wrong because it would show up on an lna run.

Reply:
I like the running list idea. Maybe something along those lines?

1. Add the filter bundles to a list of bundles to stop.
1. Run through that list until all bundles have been stopped, or at most _n_ times, moving those that were stopped to a separate list of bundles to restart. If we can't stop all bundles after _n_ tries, log an error asking the admin to restart the system and bail.
1. Run through the list of bundles to restart until all bundles have been restarted, at most _n_ times, removing those that were successfully restarted. If we can't restart all bundles after _n_ tries, log an error asking the admin to restart the system and bail.

Thoughts?